### PR TITLE
Marshal FluxConfig if present

### PR DESCRIPTION
**Pull request by danbudris**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/1826

*Description of changes:*
Make sure that we marshal flux config if it is present. 

If both a FluxConfig and a GitOpsConfig are present, use the GitOpsConfig.
This will allow us to use the FluxConfig internally while ensuring we marshal the expected cluster configuration for the user.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clustermarshaller/clustermarshaller.go [hunk 1]
```diff
@@ -28,6 +28,10 @@ func MarshalClusterSpec(clusterSpec *cluster.Spec, datacenterConfig providers.Da
 		marshallables = append(marshallables, clusterSpec.GitOpsConfig.ConvertConfigToConfigGenerateStruct())
 	}
 
+	if clusterSpec.FluxConfig != nil {
```
Comment (g-gaston) [inline pkg/clustermarshaller/clustermarshaller.go:31]: Should we have an if/else to only marshal either `GitOpsConfig` or `FluxConfig`?

Comment (g-gaston) [inline pkg/clustermarshaller/clustermarshaller.go:31]: Maybe a add a unit test fir this case?

Comment (danbudris) [reply pkg/clustermarshaller/clustermarshaller.go:35]: yeah, I added a check that throws an error if both are present, and filed at ticket to add a higher-level check for mutual exclusivity. Additionally, added a test which checks that if both are present we throw an error.

Comment (danbudris) [reply pkg/clustermarshaller/clustermarshaller.go:35]: add a unit test which tests that an error is thrown in both are present

Comment (danbudris) [reply pkg/clustermarshaller/clustermarshaller.go:35]: added a unit test for this specific case as well

**Review by g-gaston — APPROVED** [review]

**Pull request merged**